dex/networks/erc20/contracts/ERC20SwapV0.sol

```diff
@@ -18,18 +17,23 @@ pragma solidity = 0.8.6;
 // locktime. The participant can redeem at any time after the initiation
 // transaction is mined if they have the secret that hashes to the secret hash.
 // Otherwise, the initiator can refund funds any time after the locktime.
+// 
+// In order to save on gas fees, a separate ERC20Swap contract is deployed
+// for each ERC20 token. The TOKEN_ADDRESS constant must be updated for
+// each contract deployment.
 //
 // This contract has no limits on gas used for any transactions.
 //
 // This contract cannot be used by other contracts or by a third party mediating
 // the swap or multisig wallets.
-//
-// This code should be verifiable as resulting in a certain on-chain contract
-// by compiling with the correct version of solidity and comparing the
-// resulting byte code to the data in the original transaction.
 contract ERC20Swap {
     bytes4 private constant TRANSFER_FROM_SELECTOR = bytes4(keccak256("transferFrom(address,address,uint256)"));
     bytes4 private constant TRANSFER_SELECTOR = bytes4(keccak256("transfer(address,uint256)"));
+    
+    // TOKEN_ADDRESS must be updated whenever deploying a swap contract for a new token.
+    // Currently it is set to the address where the test token contract on the simnet
+    // is deployed.
+    address public constant TOKEN_ADDRESS = 0x6b4368d3E41a60e20FF8539C843B3CDB38C8A507;
```

Comment:
Can we add it as a constructor argument?

Reply:
It'll be 2k extra gas if we do that.

Reply:
Per transaction? Isn't the constructor only called once when the contract is deployed?

Reply:
Yes, but in the address is a constant, then the address will be part of the on chain bytecode instead of part of the storage, and storage lookups cost gas. It's not crazy expensive, ~$1 extra per transaction at the current prices, but it's not so complicated to handle the deployment like this either.

Reply:
Should be good with an **immutable** value, no?

https://docs.soliditylang.org/en/v0.8.10/contracts.html#constant-and-immutable-state-variables

Reply:
Ah nice, I didn't know about this. I'll try it.

Reply:
This is good to go now, the deployment to simnet works.